restapi/src/main/java/io/stargate/web/docsapi/resources/error/ErrorHandler.java

```diff
@@ -48,6 +49,10 @@ public Response apply(Throwable throwable) {
                   "Role unauthorized for operation: " + throwable.getMessage(),
                   Response.Status.UNAUTHORIZED.getStatusCode()))
           .build();
+    } else if (throwable instanceof JsonProcessingException) {
+      return Response.status(Response.Status.BAD_REQUEST)
```

Comment:
Although this will work and is good for being safe, shouldn't we catch exceptions at the place where they can occur and provide more info to the user? See `ReactiveDocumentsService` to see how I handled that so that user has a nicer feedback.

Furthermroe if the `JsonProcessingException` is caught here, then you don't know if this is related to user input or not, and you are potentially hiding a real `500` that occurred due to the bug in our code. So I would strongly advise to not do it this way.

Reply:
Yeah, I can change this up and maybe throw the appropriate `ErrorCodeRuntimeException` at each location where processing input can find malformed JSON